### ISSUE TITLE
cri multine.parser doesn't parse when tailing on multiple containers

### ISSUE THREAD

**Issue by rsumukha** [fluent/fluent-bit#3870]
## Bug Report

When multiline.parser is set to `cri` and tail plugin is configured to tail  on multiple containers logs (`/var/log/containers/*.log`), only one of the containers logs are processed by multiline.parser.

Steps to reproduce the issue
* Launch a pod on a kubernetes cluster, the pod should have more than one container.
```
pod-spec.yaml
---
 apiVersion: v1
 kind: Pod
 metadata:
   name: test-multiline
   labels:
     app: web
 spec:
   containers:
     - name: container-1
       image: busybox
       args: [/bin/sh, -c, 'while true; do echo "CONTAINER1$(head -c 17 /dev/zero | tr "\0" "+")END"; sleep 5; done']
       ports:
         - containerPort: 88
     - name: container-2
       image: busybox
       args: [/bin/sh, -c, 'while true; do echo "CONTAINER2$(head -c 17 /dev/zero | tr "\0" "+")END"; sleep 5; done']
       ports:
         - containerPort: 88
```
* Apply the fluent-bit config and start the fluent-bit process.
```
[INPUT]
    Name tail
    DB /var/log/flb_kube.db
    Mem_Buf_Limit 10MB
    Path /var/log/containers/*.log
    Read_From_Head On
    Refresh_Interval 10
    Rotate_Wait 30
    Skip_Long_Lines On
    Tag kube.*
    multiline.parser cri

[OUTPUT]
   Name stdout
   Match kube.*
```

**Expected behavior**
```
[0] kube.var.log.containers.test-multiline_default_container-2-9826d852ce1eb116cca76ea490b2efc71a23e3f938c8aabbf21a2b6135e22b4e.log: [1627072846.309327516,{"time"=>"2021-07-23T20:40:40.247781606Z", "stream"=>"stdout", "_p"=>"F", "log"=>"CONTAINER2+++++++++++++++++END"}]

[0] kube.var.log.containers.test-multiline_default_container-1-2782d81cb463ec8eaf0008f9470de6d578bafa313933a6c317b0461f99d158ec.log: [1627072846.309327516, {"time"=>"2021-07-23T20:40:46.309327516Z", "stream"=>"stdout", "_p"=>"F", "log"=>"CONTAINER1+++++++++++++++++END"}]
```

**Current behavior**
```
[0] kube.var.log.containers.test-multiline_default_container-2-9826d852ce1eb116cca76ea490b2efc71a23e3f938c8aabbf21a2b6135e22b4e.log: [0.000000000, {"log"=>"2021-07-23T20:40:40.247781606Z stdout F CONTAINER2+++++++++++++++++END"}]

[0] kube.var.log.containers.test-multiline_default_container-1-2782d81cb463ec8eaf0008f9470de6d578bafa313933a6c317b0461f99d158ec.log: [1627072846.309327516, {"time"=>"2021-07-23T20:40:46.309327516Z", "stream"=>"stdout", "_p"=>"F", "log"=>"CONTAINER1+++++++++++++++++END"}]

[0] kube.var.log.containers.test-multiline_default_container-2-9826d852ce1eb116cca76ea490b2efc71a23e3f938c8aabbf21a2b6135e22b4e.log: [0.000000000, {"log"=>"2021-07-23T20:40:47.248857219Z stdout F CONTAINER2+++++++++++++++++END"}]

[0] kube.var.log.containers.test-multiline_default_container-1-2782d81cb463ec8eaf0008f9470de6d578bafa313933a6c317b0461f99d158ec.log: [1627072853.310281793, {"time"=>"2021-07-23T20:40:53.310281793Z", "stream"=>"stdout", "_p"=>"F", "log"=>"CONTAINER1+++++++++++++++++END"}]

[0] kube.var.log.containers.test-multiline_default_container-2-9826d852ce1eb116cca76ea490b2efc71a23e3f938c8aabbf21a2b6135e22b4e.log: [0.000000000, {"log"=>"2021-07-23T20:40:54.250330776Z stdout F CONTAINER2+++++++++++++++++END"}]

```

**Your Environment**
* Version used: 1.8.2

* Environment name and versio. Kubernetes cluster 1.20

**Additional context**
* Tried to apply a pod that writes >16k chars when `max_container_log_line_size = 16384`. With multiline.parser set to cri the logs should be merged. Only one of the container logs are merged while the other was split into `P` and `F`. Found out that the second container logs aren't being parsed by multiline.parser regardless of the tags (P/F).

**Comment by edsiper** [comment]
would you please try out the following new test image ?

```
edsiper/fluent-bit:1.8.3-stream-fix-dev-92cf0
```

**Comment by rsumukha** [comment]
Tried out the new image, seeing the same result as before.

**Config used:**
```
[SERVICE]
    Flush         1
    Log_Level     info

[INPUT]
    Name tail
    DB /var/log/flb_kube.db
    Mem_Buf_Limit 10MB
    Path /test-cri/*.log
    Read_From_Head On
    Refresh_Interval 10
    Rotate_Wait 30
    Skip_Long_Lines On
    Tag kube.*
    multiline.parser cri

[OUTPUT]
    Name stdout
    Match kube.*
```

**Log Content:**
```
cat test-cri/ctr1.log test-cri/ctr2.log
2021-07-28T21:36:45.333295967Z stdout F ctr1+++++++++++++++++++++++END
2021-07-28T21:36:55.333295967Z stdout F ctr1+++++++++++++++++++++++END
2021-07-28T21:37:05.333295967Z stdout F ctr1+++++++++++++++++++++++END


2021-07-28T21:36:45.333295967Z stdout F ctr2+++++++++++++++++++++++END
2021-07-28T21:36:55.333295967Z stdout F ctr2+++++++++++++++++++++++END
2021-07-28T21:37:05.333295967Z stdout F ctr2+++++++++++++++++++++++END

```

**Output:**
```
[2021/07/28 21:40:31] [ info] [input:tail:tail.0] inotify_fs_add(): inode=48430136 watch_fd=1 name=/test-cri/ctr1.log
[2021/07/28 21:40:31] [ info] [input:tail:tail.0] inotify_fs_add(): inode=48430107 watch_fd=2 name=/test-cri/ctr2.log
[0] kube.test-cri.ctr1.log: [1627508205.333295967, {"time"=>"2021-07-28T21:36:45.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr1+++++++++++++++++++++++END"}]
[1] kube.test-cri.ctr1.log: [1627508215.333295967, {"time"=>"2021-07-28T21:36:55.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr1+++++++++++++++++++++++END"}]
[2] kube.test-cri.ctr1.log: [1627508225.333295967, {"time"=>"2021-07-28T21:37:05.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr1+++++++++++++++++++++++END"}]
[0] kube.test-cri.ctr2.log: [1627508431.035214800, {"log"=>"2021-07-28T21:36:45.333295967Z stdout F ctr2+++++++++++++++++++++++END"}]
[1] kube.test-cri.ctr2.log: [1627508431.035214800, {"log"=>"2021-07-28T21:36:55.333295967Z stdout F ctr2+++++++++++++++++++++++END"}]
[2] kube.test-cri.ctr2.log: [1627508431.035214800, {"log"=>"2021-07-28T21:37:05.333295967Z stdout F ctr2+++++++++++++++++++++++END"}]
```

**Comment by tonychoe** [comment]
I've tested using fluent/fluent-bit:1.8.3. It fixed the same issue I had with 1.8.2.

**Comment by rsumukha** [comment]
Seems like the issue has been fixed in 1.8.3. Thank you, will wait for the official release.


```
Fluent Bit v1.8.3
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/08/02 19:43:41] [ info] [engine] started (pid=1)
[2021/08/02 19:43:41] [ info] [storage] version=1.1.1, initializing...
[2021/08/02 19:43:41] [ info] [storage] in-memory
[2021/08/02 19:43:41] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/08/02 19:43:41] [ info] [cmetrics] version=0.1.6
[2021/08/02 19:43:41] [ info] [input:tail:tail.0] multiline core started
[2021/08/02 19:43:41] [ info] [sp] stream processor started
[2021/08/02 19:43:41] [ info] [input:tail:tail.0] inotify_fs_add(): inode=48430136 watch_fd=1 name=/test-cri/ctr1.log
[2021/08/02 19:43:41] [ info] [input:tail:tail.0] inotify_fs_add(): inode=48430107 watch_fd=2 name=/test-cri/ctr2.log
[0] kube.test-cri.ctr1.log: [1627508205.333295967, {"time"=>"2021-07-28T21:36:45.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr1+++++++++++++++++++++++END"}]
[1] kube.test-cri.ctr1.log: [1627508215.333295967, {"time"=>"2021-07-28T21:36:55.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr1+++++++++++++++++++++++END"}]
[2] kube.test-cri.ctr1.log: [1627508225.333295967, {"time"=>"2021-07-28T21:37:05.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr1+++++++++++++++++++++++END"}]
[0] kube.test-cri.ctr2.log: [1627508205.333295967, {"time"=>"2021-07-28T21:36:45.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr2+++++++++++++++++++++++END"}]
[1] kube.test-cri.ctr2.log: [1627508215.333295967, {"time"=>"2021-07-28T21:36:55.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr2+++++++++++++++++++++++END"}]
[2] kube.test-cri.ctr2.log: [1627508225.333295967, {"time"=>"2021-07-28T21:37:05.333295967Z", "stream"=>"stdout", "_p"=>"F", "log"=>"ctr2+++++++++++++++++++++++END"}]

```

**Comment by agup006** [comment]
Great to hear! I'll close this issue out
